esmvaltool/utils/testing/regression/compare.py

```diff
@@ -0,0 +1,398 @@
+"""Compare recipe runs to previous runs."""
+from __future__ import annotations
+
+import argparse
+import difflib
+import filecmp
+import fnmatch
+import os
+import sys
+from pathlib import Path
+from textwrap import indent
+from typing import Iterator, Optional
+
+import numpy as np
+from PIL import Image
+import xarray as xr
+
+try:
+    import imagehash
+except ImportError:
+    print("Please `pip install imagehash`")
+
+IGNORE_FILES: tuple[str, ...] = (
+    '*_citation.bibtex',
+    '*_data_citation_info.txt',
+    '*_provenance.xml',
+    'metadata.yml',
+)
+"""Files to ignore when comparing results."""
+
+IGNORE_GLOBAL_ATTRIBUTES: tuple[str, ...] = (
+    'creation_date',
+    'history',
+    'provenance',
+    'software',
+)
```

Comment:
I think it is a mistake in the diagnostic to attach the `software` and `history` attributes to the variable instead of as global attributes. For example, it seems quite unlikely that two variables in the same file will have been created by differing versions of the software.

Regarding the attributes `ncdf`, `filename`, `ancestors` and `source_file`: why are these set? We have a quite advanced system in place for tracking provenance, so why do diagnostics try to implement their own crude form of provenance tracking? I think such a feature can simply be removed.